### DISCUSSION TITLE
Don't understand the inferencer's behavior

### DISCUSSION THREAD

**Discussion by nahuelranieri** [refinedev/refine#5341] [Q&A] [answered]
I am discovering the refine tool and on paper I quite like the idea of inferencer. I understand that it makes a _guess_ and generates some code. But I keep gettin the error "Something went wrong while fetching the resource data" using apis with the same structure as the one provided in the documentation with api.fake-rest. In this example I used the very famous pokeapi:
`const API_URL = "https://api.fake-rest.refine.dev";
 const POKEMON_API_URL = "https://pokeapi.co/api/v2/pokemon/";`

Then in the _Refine_ component pass the following in the data Provider's prop:
`dataProvider={{
                  default: dataProvider(API_URL),
                  pokemon: dataProvider(POKEMON_API_URL),}}`
and in resources:
```
resources={[
                  {
                    name: "blog_posts",
                    //the crud for blog example
                    },
                  },
                  {
                    name: "pokemon",
                    list: PokemonList,
                    meta: {
                      canDelete: true,
                      dataProviderName: "pokemon",
                    },
                  },]}
```
`<Route path='/pokemon'>
                      <Route index element={<PokemonList/>}/>
</Route>`
Finally, in the PokemonList component I use the example structure, with the MuiInferencer.

I understand that I can create a custom dataProvider but as the idea would be to use refine in several projects, and in some of which I will use different apis, I would like to know if there is a simpler way to solve the use of MuiInferencer or understand what I am doing wrong.

Thank you in advance

**Comment by aliemir** [comment, marked as answer]
Hey @nahuelranieri, sorry for the issue! I had to admit that this error message is not really explaining much of the issue, sorry for that 🤦

Does the error message also have the resource name? If so, then it's related with the first fetch that is sent for the main record data.

If it doesn't have the resource name, then it's because there's an issue with the relation requests. In this case, you should be able to see the generated code using the prompt at the bottom of the screen and we can investigate it further if you can share the generated output and provide a sample response from the API.

I'll also check this myself and try to reproduce the issue. Thank you for reporting this 🙏 I'm also adding a task to our to-do to update error messages of the inferencer components to provide meaningful traces. 🚀
